### PR TITLE
ci: bump ci image to ubuntu 24.04

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: scripts/source/prepare_ci_kafka.sh [hunk 1]
```diff
@@ -3,13 +3,6 @@
 # Exits as soon as any line fails.
 set -e
 
-KCAT_BIN="kcat"
-# kcat bin name on linux is "kafkacat"
-if [ "$(uname)" == "Linux" ]
-then
-    KCAT_BIN="kafkacat"
-fi
-
```
Comment (xxchan) [inline scripts/source/prepare_ci_kafka.sh:12]: fun fact: Actually this is my only original motivation for this PR...

**Comment by xxchan** [comment]
@lmatz This doesn't close #16139. It only changes `rw-build-env`, instead of `risingwave` 🤣

**Review by BugenZhao — APPROVED** [review]

**Comment by lmatz** [comment]
oh, I thought it is RW's `Dockerfile`

**Review by wangrunji0408 — APPROVED** [review]

**Comment by lmatz** [comment]
one thing I am confused about, the current 24.04 is not an LTS version right?

**Comment by xxchan** [comment]
> one thing I am confused about, the current 24.04 is not an LTS version right?

I think "LTS" just means the (major) version will be supported in long time. Current version is beta.

**Pull request merged**